### DISCUSSION TITLE
A more in-depth MRE tutorial

### DISCUSSION THREAD

**Discussion by yeppiidev** [raspiduino/mre-sdk#2] [Ideas]
If possible & interested, can you do a more in-depth on how to set up MRE SDK and how to create a simple app? I know there's a repo by UstadMobile, but it seems like it's the only repo that actually talks about "MRE". Extending to that information might be useful in the future.

**Comment by raspiduino** [comment]
Sorry for late reply, I had a busy week at FRC :)

Alright, I understand that this platform is not properly documented (if I don't say it's almost no document at all, like other Chinese commercial softwares :)))

Ok, I know it will be really long, but I will still do it :)

It might takes a while for writing the full tutorial, so probably I will write it in parts and put it here

Thank you

**Reply by yeppiidev** [reply]
i'll transfer the contents of this page/discussion to my personal page http://jmeapps.wap.sh/ so that this information can be accessed from an MRE device 👍🏼

**Reply by yeppiidev** [reply]
maybe consider transferring the contents from this discussion to the wiki tab too i guess

**Comment by raspiduino** [comment]
# Step 1: Setup the development environment

Quick summary:
1. Download and install MRE SDK
2. Download and install a C/C++ compiler (recommended) or (possibly) any compilers that can output machine code for ARM(7TDMI)
3. (Optional) Download and install Visual Studio with WinSDK and C++ development package for debugging the app in MODIS (kinda (but not really) MRE simulator on PC)

## 1. Download and install MRE SDK
Go to [Releases](https://github.com/raspiduino/mre-sdk/releases) in this repo to download the SDK.
<br> The installation process is really simple, so I will not talk about that here. It's just the same as other installers. Just follows instructions on the screen.

2. Download and install a compiler
If you got Nokia devices running on ARM platform, it's really likely that you have a Mediatek SoC based on ARM7TDMI processor, therefore you should download a compiler for that architecture. My recommend: [GNU Arm Embedded Toolchain 4.8](https://launchpad.net/gcc-arm-embedded/4.8/4.8-2014-q3-update)

If you want to use other language / compiler, then that's OK, as long as you can output ARM7TDMI machine code in ELF file format. If you use languages other than C / C++ or assembly, then there will be no support by SDK. You have to bind the API yourself.

If you have devices other than Nokia running MRE, then you will have to download proper compiler for your platform.
